compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -13633,8 +13633,7 @@ private STNode parseTransactionStatement(STNode transactionKeyword) {
         startContext(ParserRuleContext.TRANSACTION_STMT);
         STNode blockStmt = parseBlockNode();
         endContext();
-        STNode onFailClause = parseOptionalOnFailClause();
-        return STNodeFactory.createTransactionStatementNode(transactionKeyword, blockStmt, onFailClause);
+        return STNodeFactory.createTransactionStatementNode(transactionKeyword, blockStmt);
```

Comment:
We can remove the optional `onFailClause` in the ST also right?
https://github.com/ballerina-platform/ballerina-lang/blob/2d730a699ebe7ff6be930cfe7e47a2b09c38473c/compiler/ballerina-treegen/src/main/resources/syntax_tree_descriptor.json#L3595-L3599

Reply:
Seems the changes to `syntax_tree_descriptor.json` is not committed. Also when we change it we have to update `api_gen_syntax_tree_descriptor.json` as well.

Reply:
Updated. 6e28d27522656155f4d8712f60a34356460eecca